src/item.cpp

```diff
@@ -2682,96 +2682,118 @@ void item::final_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
-                        } else if( sizing_level == sizing::big_sized_human_char ||
-                                   sizing_level == sizing::big_sized_small_char ) {
-                            resize_str = _( "<bad>can not be downsized</bad>" );
-                        } else if( sizing_level == sizing::small_sized_big_char ||
-                                   sizing_level == sizing::human_sized_big_char ) {
-                            resize_str = _( "<bad>can not be upsized</bad>" );
-                        }
-                        std::string info_str = string_format( _( "* This clothing %s." ), resize_str );
-                        info.push_back( iteminfo( "DESCRIPTION", info_str ) );
+                    switch( sizing_level ) {
+                        case sizing::small_sized_human_char:
+                            info.emplace_back( "DESCRIPTION",
+                                               _( "* This clothing <info>can be upsized</info>." ) );
+                            break;
+                        case sizing::human_sized_small_char:
+                            info.emplace_back( "DESCRIPTION",
+                                               _( "* This clothing <info>can be downsized</info>." ) );
+                            break;
+                        case sizing::big_sized_human_char:
+                        case sizing::big_sized_small_char:
+                            info.emplace_back( "DESCRIPTION",
+                                               _( "* This clothing <bad>can not be downsized</bad>." ) );
+                            break;
+                        case sizing::small_sized_big_char:
+                        case sizing::human_sized_big_char:
+                            info.emplace_back( "DESCRIPTION",
+                                               _( "* This clothing <bad>can not be upsized</bad>." ) );
+                            break;
+                        default:
+                            break;
                     }
                 } else {
-                    std::string resize_str;
-                    if( sizing_level == sizing::small_sized_human_char ) {
-                        resize_str = _( " and <info>upsized</info>" );
-                    } else if( sizing_level == sizing::human_sized_small_char ) {
-                        resize_str = _( " and <info>downsized</info>" );
-                    } else if( sizing_level == sizing::big_sized_human_char ||
-                               sizing_level == sizing::big_sized_small_char ) {
-                        resize_str = _( " but <bad>not downsized</bad>" );
-                    } else if( sizing_level == sizing::small_sized_big_char ||
-                               sizing_level == sizing::human_sized_big_char ) {
-                        resize_str = _( " but <bad>not upsized</bad>" );
+                    switch( sizing_level ) {
```

Comment:
I just went through the trouble of simplifying this code with a generated string; is there any particular reason you threw that out in favor of repeated descriptions?

Reply:
Well, it *was* to avoid calling `push_back` in the default case, but in retrospect, I really should have just put in a check `!resize_str.empty()`.

Reply:
Well, it *was* to avoid calling `push_back` in the default case, but in retrospect, I really should have just put in a check `!resize_str.empty()`.

Just did so, and reverted to the way you were handling string generation.